spec/useLoadClientScript.spec.tsx

```diff
@@ -1,11 +1,52 @@
 import { describe, expect, it } from "vitest"
 import { renderHook } from "@testing-library/react"
 import { useLoadClientScript } from "../src/hooks"
+import "@testing-library/jest-dom/vitest"
+
+function loadClientScript(merchant: string) {
+  const script = document.createElement("script")
+  script.setAttribute("nosto-client-script", "")
+  script.src = `https://connect.nosto.com/include/${merchant}`
+  script.type = "text/javascript"
+  script.async = true
+  const promise = new Promise<void>(resolve => {
+    script.onload = () => {
+      window.nosto?.reload({ site: "localhost" })
+      resolve()
+    }
+  })
+  document.body.appendChild(script)
+  return promise
+}
 
 describe("useLoadClientScript", () => {
-  it("set loaded state to true when client is loaded externally", () => {
-    const { result } = renderHook(() => useLoadClientScript({ loadScript: false, account: "shopify-11368366139" }))
-    expect(result.current.clientScriptLoaded).toBe(true)
+
+  it("loads client script", async () => {
+    const hook = renderHook(() => useLoadClientScript({ account: "shopify-11368366139" }))
+    await new Promise(window.nostojs)
+
+    hook.rerender()
+    expect(hook.result.current.clientScriptLoaded).toBe(true)
+    expect(window.nosto).toBeDefined()
+  })
+
+  it("set loaded state to true when client is loaded externally after", async () => {
```

Comment:
What about adding a check if this test adds two Nosto script tags at some point on the rendered page?

Reply:
will do